### PR TITLE
Rails 6 Support

### PR DESCRIPTION
**Pull request by thiagopradi**
Hi Folks,

This PR adds supports for Rails 6. A few things that I did:

- Removed Rails 3/4 from the build matrix (they have reached EOL already)
- Updated Travis config file
- Fixes SQLite version

This is still WIP - I need to check if all gems are working correclty on Rails 6.

Thiago

### CODE REVIEW

**Comment by mtchavez** [comment]
Any chance the maintainers can chime in here? Sounds like there might need to be a major version bump since rails 3 and 4 support is moved. I can help do any testing of this as well.

**Comment by theaxel** [comment]
When can we expect this to be merged @domaindrivendev? Anything holding it back from being merged?

File: .ruby-version
Comment (sumitdey035) [inline .ruby-version]: The latest Ruby version is 2.6.4, so can we update it here?

**Review by BookOfGreg — APPROVED** [review]

**Review by Asuforce — APPROVED** [review]

**Comment by phoet** [comment]
looks like the ruby community is out of office...

**Comment by wwahammy** [comment]
It looks like @domaindrivendev is having trouble keeping up (which is not a criticism!)

Since this gem has a lot of interest, maybe it'd make sense to fork this and reorganize on another repo? Additionally, we should develop a more sustainable governance model (with multiple maintainers) so we dont put so much work on a single maintainer.

If you're willing to get involved and participate in that, please respond to this!

File: .ruby-version
Comment (thiagopradi) [reply .ruby-version]: Updated - thanks for the review!

**Comment by thiagopradi** [comment]
@domaindrivendev - just updated the PR with the latest ruby version. Let me know if you want some extra help testing this PR before merging it.

**Comment by BookOfGreg** [comment]
@thiagopradi Can I ask you to open this PR against https://github.com/rswag/rswag ?

@domaindrivendev I opened an organization for rswag at https://github.com/rswag/rswag of which I invited you as a member. You can be the owner of it if you want, I just wanted to get the ball rolling!
I am the sole maintainer of https://github.com/reactjs/react-rails so I am trustworthy in-case you consider this a bit left-field. My email is in my profile if you wish to discuss anything with me privately.

@wwahammy I have opened up an organization. We will see if it takes off or not. Thanks for the idea, would you like to contribute to it there?

**Comment by thiagopradi** [comment]
@BookOfGreg - of course! my week will be pretty busy - but I will try to open it ASAP.

**Comment by thiagopradi** [comment]
@BookOfGreg - PR opened: https://github.com/rswag/rswag/pull/2

**Comment by domaindrivendev** [comment]
@BookOfGreg and all other folks on this thread - apologies for my lack of responsiveness. It's great to see the level of interest in this project but, as you've probably gathered, I don't have any time to develop it further or even keep up with basic maintenance.

Therefore, I'm happy to see it transition to an organization with a more community focused maintenance model. Thanks for taking the initiative on this.

If there's any quick changes I can make with this repo to assist (e.g. README banner, automatic redirect to the new repo etc.), just let me know and I'll make it happen - so long as they don't require me to have a Ruby dev environment up and running 😉 (unfortunately I'm doing a lot of .NET work on Windows these days)

**Comment by BookOfGreg** [comment]
@domaindrivendev Seems I was a bit quick to fork. In order to transfer ownership I will first delete rswag/rswag, by the time you read this I will have already done that (sorry thiagopradi, we will re-merge this 😆 )
I can re-do the differences between them easy enough.
Remember to accept member or owner privilege on the rswag organization.

Also thank you so much for making this tool. I do dotnet core development too and use swashbuckle also, so huge thanks for bringing swagger to both communities. We will keep up the progress here.

**Comment by pedantic-git** [comment]
Thanks for all your effort to bring this into an org! As an aside: Are you aware of @jdanielian's excellent fork https://github.com/jdanielian/open-api-rswag that brings rswag up to OpenAPI 3.0? Might be worth bringing that fork into the new org too?

**Comment by BookOfGreg** [comment]
Part of bringing over a gem is looking at all the forks for things to pull in to the contribution. That's great to know the fork for OpenAPI 3 is out there, I'll definitely skim through looking for other forks to pull in.

**Comment by gionaufal** [comment]
Just for you to know, looks like this breaks assets loading for Rails 6 using API only mode

**Comment by BookOfGreg** [comment]
@domaindrivendev I've deleted the fork of rswag over at the rswag organization.
What would you like to do next? You can transfer the gem into the org. I could also transfer the org to you first if you would prefer that.

You may also wish to consider how to automate gem releases, or share release access so that those can continue smoothly. I'd consider either automation around publishing releases in Github triggering, or automation around git tags, or allowing another maintainer on the gem within rubygems.

I'm happy to spend some time working on the gem as part of office hours for the next short while as we're looking to upgrade to Rails 6 + Swagger 3 so if we choose to move on this through this week then I have a small team that could contribute.

**Comment by domaindrivendev** [comment]
@BookOfGreg I just transferred the repo over to the `rswag` organization. I don't have to be the owner of the organization, but I'd like to have full admin access if that's possible. If making me the owner is the easiest way to achieve that then I'm ok with that too.

Regarding automated gem releases, that's already setup via travis ci based on git tags. IIRC, any push to origin will trigger a full test pass (unit and integration tests). Additionally, if a new tag is pushed, then the component gems (`rswag-specs`, `rswag-api` and `rswag-ui`) _and_ the master gem (`rswag`) will be automatically pushed to rubygems (assuming all the tests pass).

The only real ask I'd have is when releasing going forward, please adhere to semantic versioning. Thanks, and let me know if this is everything you need to give the project this lease of new life? ❤️

**Comment by BookOfGreg** [comment]
image
You are now owner level on the organization.

Wonderful that the git tags are still setup to release. This evening I'll ensure that the Travis builds are still hooked up OK and merge this PR, tweak the Readme and maybe look at a point release for Rails 6 support?
I need to do a release for react-rails so might as well make a night of it :)

Now we have the organization you can still reach me via email (public on profile) or now use the member discussion pages on the org as well as the normal public channels. (such as @thiagopradi 's PR 😆 )
As far as plan to revive the gem somewhat, you still know the internals better than likely anyone else so I'd possibly say opening issues for potential refactorings or ideas beyond the obvious (OpenAPI 3 support) would be handy as well as keeping an eye open for others who might want to become regular contributors to the org.

File: rswag-ui/rswag-ui.gemspec
Comment (BookOfGreg) [inline rswag-ui/rswag-ui.gemspec:17]: I'll move the pessimistic version pin to '< 7.0' in the future.

**Pull request merged**
